OpenSim/Common/Test/testComponentInterface.cpp

```diff
@@ -918,6 +918,12 @@ void testListSockets() {
 
 void testComponentPathNames()
 {
+    Foo foo;
+    foo.setName("LegWithConstrainedFoot/foot");
```

Comment:
Might want to check a name like `"a\b"` as well to make sure the character is properly escaped and captured. Could additionally add a check for `"a:b"` that doesn't currently throw, with a TODO for later.

Reply:
I will have to include when addressing #2007, so will leave it for now if OK with you.

Reply:
Seems like checking "`\`" would belong in this PR since all `invalidChars = "\\/*+ \t\n"` should now be caught, but ok to include later. (I expected to see a test for each of these invalid characters since it wouldn't cost many lines to test each one, e.g., using a lambda.) I've added a reminder to #2007.